pontoon/base/static/js/fluent_interface.js

```diff
@@ -340,19 +340,17 @@ var Pontoon = (function (my) {
        * Message is supported if all value elements
        * and all attribute elements are of type:
        * - TextElement
-       * - ExternalArgument
-       * - MessageReference
-       * - SelectExpression?
+       * - Placeable with expression type ExternalArgument, MessageReference or SelectExpression
        */
       isSupportedMessage: function (ast) {
+        var self = this;
+
         function elementsSupported(elements) {
-          return elements.every(function(item) {
-            return [
-              'TextElement',
-              'ExternalArgument',
-              'MessageReference',
-              'SelectExpression'
-            ].indexOf(item.type) >= 0;
+          return elements.every(function(element) {
+            return (
+              self.isSimpleElement(element) ||
+              (element.expression && element.expression.variants)
```

Comment:
`variants` is always a non-empty array in `SelectExpression`. Did you mean to test for the selector? That would be `element.expression.expression`.

Reply:
(It's a bit weird that it's called `expression` rather than `selector`).

Reply:
I'm checking if the element has variants, which effectivelly means if it's a `SelectExpression`. I could be more explicit and say `element.expression.type === 'SelectExpression'`.

But I use the same check on several occasions where `element.expression.variants` is used if the check passes, so I found the code easier to understand that way - otherwise you'd need to know that if the element is of type `SelectExpression`, it must have variants.

I think I should factor this out into something like `isSelectExpression{Element/Placeable}`, use the existing check and add a comment describing this. Thoughts?

And I agree that `element.expression.expression` is confusing. :)